core/api/src/main/java/org/trellisldp/api/ResourceService.java

```diff
@@ -136,8 +137,7 @@ default RDFTerm unskolemize(final RDFTerm term) {
      */
     default Stream<? extends Quad> export(final Collection<IRI> graphNames) {
         return scan().map(Triple::getSubject).filter(x -> x instanceof IRI).map(x -> (IRI) x)
-            // TODO - JDK9 optional to stream
-            .flatMap(id -> get(id).map(Stream::of).orElseGet(Stream::empty))
+            .parallel().map(this::get).map(CompletableFuture::join)
```

Comment:
Do we need `::join` here? Can we not let the parallel threads produce results as they can?

Reply:
If we continue to return a `Stream<? extends Quad>`, we'll need a `join` somewhere in the chain operations. Performance-wise, it is quite likely that deferring the `join` until the end of the sequence would work. That said, the `export` method is also a bit of a speculative method, given that it is not ever called in the HTTP layer. The idea is to make it available as part of a CLI utility (or some future HTTP endpoint). So until we have an actual use for this (and some data against which to test), I'd be inclined to either leave this as-is or else remove the method entirely; we can always add it back later if there is a use for it.

Reply:
Dumping all resources at one resource seems profoundly nonRESTful. That kind of work should be taking place on the back end. Let's remove this unless we have a reason for it?

Reply:
:+1: